rust/src/mqtt/mqtt.rs

```diff
@@ -41,6 +42,14 @@ static mut MQTT_MAX_TX: usize = 1024;
 
 static mut ALPROTO_MQTT: AppProto = ALPROTO_UNKNOWN;
 
+#[derive(AppLayerFrameType)]
+pub enum MQTTFrameType {
+    Pdu,
+    Header,
+    Data,
+    TruncData,
```

Comment:
Following the discussion that happened privately, this data frame type does not seem to be necessary as we cannot annotate something that does not exist. Even if the data is truncated, the parsed parts can be tagged with `Pdu`, so, we can get rid of this. Make sure to add tests corresponding to truncated data. e.g.

```
aaaaaaaaaaaaaaabbbbbbbbbb 
```
where `a..a` == data exactly the length limit
`b..b` == extra data that should ideally be truncated by the parser.
So, you have three tests.
1. Well within the boundary.
check for say first 4B of data in `Pdu`, it should match.
2. Boundary test. Should match.
check for all `aaaa` exactly up to the length limit.
3. Post boundary test. Should not match.
check for `aaaa...bbb` i.e. good data + truncated data